src/array.rs

```diff
@@ -1,25 +1,26 @@
+use std::fmt::Debug;
 use std::ops::{Deref, DerefMut};
 
-#[derive(Copy, Clone)]
-pub struct Array<T, const N: usize> {
+#[derive(Copy, Clone, Debug)]
+pub struct Array<T: Debug, const N: usize> {
```

Comment:
What's the reason for requiring Debug on Array. Isn't the automatic derive implementation enough. If not, then only require the debug type constraint when implementing debug for array

Reply:
Ah yeah, not needed, I went overboard here :)